pdf_rendering_service/documents/views.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework import status
+from rest_framework.generics import GenericAPIView, RetrieveAPIView
+from rest_framework.parsers import FileUploadParser
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from pdf_rendering_service.documents.models import Document
+from pdf_rendering_service.documents.renderers import PNGRenderer
+from pdf_rendering_service.documents.serializers import (
+    CreatedDocumentSerializer,
+    DocumentSerializer,
+)
+from pdf_rendering_service.documents.services.document_store import (
+    get_document_page,
+    store_document,
+)
+from pdf_rendering_service.documents.tasks import process_document
+
+
+class DocumentUploadView(APIView):
+    parser_classes = (FileUploadParser,)
+    permission_classes = (AllowAny,)
+
+    def post(self, request, filename: str = ""):
+        file = request.data["file"]
+        document = store_document(file)
+        process_document.send(document.pk)
+        return Response(
+            data=CreatedDocumentSerializer(document).data,
+            status=status.HTTP_201_CREATED,
+        )
+
+
+class DocumentView(RetrieveAPIView):
```

Comment:
In the most cases yes.
But DocumentUploadView requires using the FileUploadParser which leads me to using extra CBS just for this operation.
Merging the views together could make the code of CBV more complex.

furthermore as the DRF docs says:
> Since this parser's media_type matches any content type, FileUploadParser should generally be the only parser set on an API view.

Reply:
Seems like we should dig into this a bit as our code would need some refactoring. 👍 
Thanks for the explanation